content/5_Plat/HLD.mdx

```diff
@@ -252,7 +252,321 @@ set problems where $\Theta(Q\sqrt N\log N)$ is intended ...
 	</Resource>
 </Resources>
 
-<IncompleteSection />
+
+### Implementation 1 - Basic Heavy-Light Decomposition
+In practical implementation, HLD begins with representing the tree using adjacency lists or matrices `adj`, establishing parent-child relationships crucial for traversal and manipulation, and initializing several essential data structures. These include arrays for storing the parent of each node `par`, identifying the heavy child for each node `heavy`, storing the depth of each node `depth`, marking the root of the heavy path to which each node belongs `root`, and mapping nodes to their positions in the Segment Tree `pos`. Additionally, a [Segment Tree](https://usaco.guide/gold/PURS?lang=cpp) `st` is initialized to handle range maximum queries and point updates effectively.
+
+The next step involves performing a Depth-First Search (DFS) starting from the root node to calculate the size of each subtree. During this traversal, the heavy child for each node is identified as the child with the largest subtree size. This is crucial because heavy paths are constructed using these heavy children. The DFS also records the depth of each node and its parent, which are essential for the subsequent steps in the decomposition process.
+
+Once the DFS is complete, the decomposition of the tree into heavy paths and light edges begins. By iterating through all nodes, we determine which nodes are the roots of new heavy paths. If a node is not the heavy child of its parent, it signifies the start of a new heavy path. From each heavy path root, we traverse down through the heavy children, marking the entire path. Each node in a heavy path is assigned a position in the segment tree, ensuring that heavy paths are continuous in the segment tree.
```

Comment:
sooo are you doing a dfs for depth and another dfs for heavy/light edges? the explanation and ordering here is weird

Reply:
It is working in order only dfs(1, 1 adj) calculates the depths and other things, and then only it moves to the decomposition part for further process in the constructor.

Reply:
thats not clear